gnomad_qc/v4/create_release/create_de_novo_release.py

```diff
@@ -42,304 +31,210 @@
+            & hl.any(lambda a: hl.is_snp(mt.alleles[0], a), mt.alleles[1:])
+        ),
     )
-    meta_ht = filter_to_trios(meta_ht, fam_ht)
-    meta_ht = meta_ht.repartition(100).checkpoint(
-        new_temp_file("releasable_trios", "ht")
+    mt = hl.experimental.sparse_split_multi(mt).checkpoint(
+        dense_trio_mt(releasable=True, split=True, test=test).path, _read_if_exists=True
     )
-
-    # Get the split gnomAD VDS filtered to high quality releasable trios and de novo
-    # variants.
-    vds = get_gnomad_v4_vds(
-        split=True,
-        high_quality_only=True,
-        chrom="chr20" if test else None,
-        filter_samples_ht=meta_ht,
-        filter_variant_ht=var_ht,
-        checkpoint_variant_data=True,
+    mt = annotate_adj(mt)
+
+    # V3 and v4 VDS have the PL and AD fields for homref genotypes intentionally
+    # removed to save storage space and costs. We need to approximate the
+    # AD and PL fields when missing.
+    mt = mt.annotate_entries(
+        AD=hl.or_else(mt.AD, [mt.DP, 0]),
+        PL=hl.or_else(mt.PL, [0, mt.GQ, 2 * mt.GQ]),
     )
-
-    return hl.vds.to_dense_mt(vds)
-
-
-# TODO: Change when we decide on the final de novo data to release.
-def custom_de_novo_select(ht, **_):
-    """Select fields from de novo stats Table."""
-    return {
-        f"de_novo_stats{'' if g == 'adj' else '_raw'}": hl.struct(
-            n_de_novo=ht[f"n_de_novos_{g}"],
-            parents=hl.struct(
-                AC=ht[f"ac_parents_{g}"],
-                AN=ht[f"an_parents_{g}"],
-            ),
-            children=hl.struct(
-                AC=ht[f"ac_children_{g}"],
-                AN=ht[f"an_children_{g}"],
-            ),
+    mt = mt.annotate_rows(prior=priors_ht[mt.row_key].freq[0].AF)
+    mt = mt.select_entries("GT", "AD", "DP", "GQ", "PL", "adj")
+
+    tm = hl.trio_matrix(mt, ped, complete_trios=True)
+    tm = tm.transmute_cols(is_xx=tm.is_female)
+    tm = tm.checkpoint(new_temp_file("trio_matrix", "mt"))
+
+    ht = tm.entries()
+
+    ht = ht.annotate(
+        de_novo_call_info=default_get_de_novo_expr(
+            locus_expr=ht.locus,
+            alleles_expr=ht.alleles,
+            proband_expr=ht.proband_entry,
+            father_expr=ht.father_entry,
+            mother_expr=ht.mother_entry,
+            is_xx_expr=ht.is_xx,
+            freq_prior_expr=ht.prior,
         )
-        for g in ["adj", "raw"]
-    }
-
-
-def custom_joint_freq_select(ht: hl.Table, **_) -> Dict[str, hl.expr.Expression]:
-    """
-    Select joint freq fields for release.
-
-    :param ht: Joint freq Hail Table.
-    :return: Select expression dict.
-    """
-    selects = {
-        f"{data_type}_freq": ht[data_type].freq
-        for data_type in ["exomes", "genomes", "joint"]
-    }
-    selects["exomes_qual_hists"] = ht.exomes.histograms.qual_hists
+    )
 
-    return selects
+    ht = ht.filter(ht.de_novo_call_info.is_de_novo).naive_coalesce(1000)
```

Comment:
should this 1000 be an argument rather than hard coded? not sure how strict we've been about this in gnomad_qc